lib/puppet_x/sqlserver/sql_connection.rb

```diff
@@ -90,25 +90,26 @@ def win32_exception
     end
 
     class ResultOutput
+
       attr_reader :exitstatus, :error_message, :raw_error_message
 
-      def initialize(has_errors, error_message)
+      def initialize(has_errors, error_message, connection)
         @exitstatus = has_errors ? 1 : 0
-        if error_message
-          @raw_error_message = error_message
-          @error_message = parse_for_error(error_message)
+
+        unless connection.nil?
+          unless connection.Errors.count == 0
+            @error_message = connection.Errors(0).Description
+          end
+        end
+
+        unless (!@error_message.nil?) && (error_message.nil?)
```

Comment:
The `unless` + negation is a bit confusing to read, especially when also combined with the `&&`.

I think your intent here is to set `@error_message` ONLY when it's not already set.

If that's correct, I think this can this be simplified like:

```ruby
@error_message ||= error_message
```

Reply:
I was looking for a more elegant want to do this. Changing now.